lib/request.js

```diff
@@ -241,7 +259,7 @@ module.exports.agent = TestAgent;
  * @api private
  */
 
-function Test (app, method, path) {
+function Test (app, method, path, onFinish) {
```

Comment:
What's `onFinish` used for?

Reply:
I believe he was going to use it as a callback, but ended up going with the `on('end', function() {` approach.
Anyways it looks like this parameter is not being used and should be removed 😄

Reply:
Oops 😊 yeah my original implementation had `onFinish` but then I realised I can use `.on`. Will remove.